flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandler.java

```diff
@@ -290,43 +289,20 @@ private void decodeMsg(Object msg) throws Throwable {
 	}
 
 	private void decodeBufferOrEvent(RemoteInputChannel inputChannel, NettyMessage.BufferResponse bufferOrEvent) throws Throwable {
```

Comment:
The above 252 line would cause NPE issue as I mentioned the concern https://github.com/apache/flink/pull/7368#pullrequestreview-354930094